aten/src/ATen/NestedTensorImpl.h

```diff
@@ -13,15 +13,32 @@ namespace native {
 struct TORCH_API NestedTensorImpl : public c10::TensorImpl {
   explicit NestedTensorImpl(at::Tensor buffer, at::Tensor nested_size_tensor);
 
+  // Note: in current implementation,
+  // empty nested tensor = no underlying tensor
+  // nesting empty tensors is not considered empty nested tensor
+  // TODO: for now, we use `nested_size_tensor_.dim()`
+  // to determine whether a nested tensor is empty
+  // when empty, `nested_size_tensor_.dim() = 0`
+  // otherwise `nested_size_tensor_.dim() = 2`
+  // maybe there is a better indicator for emptiness?
```

Comment:
numel() can also be useful here

Reply:
yeah regarding numel I'm a bit confused, since there are 2 cases where `buffer.numel() = 0`:
1. `torch.nested_tensor([])`
2. `torch.nested_tensor([torch.tensor([]), torch.tensor([])])`

But it seems to me that only case 1 is considered empty nested tensor

Reply:
@YifanShenSZ - correct. In the same way you can have a Tensor of shape `(2, 0)` with 0 elements.